### DISCUSSION TITLE
Make Error after generating Makefile with CMake choosing "Unix Makefiles" generator

### DISCUSSION THREAD

**Discussion by GFMicroPort** [sofa-framework/sofa#3696] [Build / Config / Environment] [answered]
Hi,

I am trying to install SOFA 22.06 from sources on a Windows 10 computer and I managed to configure and generate the Makefile using CMake (specifying **-G "Unix Makefiles"** in the **cmake** command), but I am now encountering an error when running the **make** command.
In fact, I obtained the following error when running **make** in an elevated PowerShell:

    [  0%] Building CXX object Sofa/framework/Config/CMakeFiles/Sofa.Config.dir/src/sofa/initSofaConfig.cpp.obj
    c++.exe: warning: /MP: linker input file unused because linking not done
    c++.exe: error: /MP: linker input file not found: No such file or directory
    c++.exe: warning: /wd4250: linker input file unused because linking not done
    c++.exe: error: /wd4250: linker input file not found: No such file or directory
    c++.exe: warning: /wd4251: linker input file unused because linking not done
    c++.exe: error: /wd4251: linker input file not found: No such file or directory
    c++.exe: warning: /wd4275: linker input file unused because linking not done
    c++.exe: error: /wd4275: linker input file not found: No such file or directory
    c++.exe: warning: /wd4675: linker input file unused because linking not done
    c++.exe: error: /wd4675: linker input file not found: No such file or directory
    c++.exe: warning: /wd4661: linker input file unused because linking not done
    c++.exe: error: /wd4661: linker input file not found: No such file or directory
    make[2]: *** [Sofa/framework/Config/CMakeFiles/Sofa.Config.dir/build.make:77: Sofa/framework/Config/CMakeFiles/Sofa.Config.dir/src/sofa/initSofaConfig.cpp.obj] Error 1
    make[2]: *** Deleting file 'Sofa/framework/Config/CMakeFiles/Sofa.Config.dir/src/sofa/initSofaConfig.cpp.obj'
    make[1]: *** [CMakeFiles/Makefile2:4859: Sofa/framework/Config/CMakeFiles/Sofa.Config.dir/all] Error 2
    make: *** [Makefile:136: all] Error 2

If someone has an idea of what could be causing this problem and on how to solve it, it will be very helpful.

Thank you very much for your help,

**Comment by fredroy** [comment, marked as answer]
GCC toolchains are not really supported on WIndows. CMake files usually assume you that if you are on Windows, then you use Visual Studio/MSVC (hence the fact that gcc does not recognize specific msvc compilation flags like `/wd4251` etc).
Though there is an experimental support for Visual Studio/clang toolchain (not tested at all)

**Reply by GFMicroPort** [reply]
Hello Frederick,
Thank you for your answer, it helped me better understand the error. Maybe I will retry the compilation using Visual Studio/MSVC since it seems to be more adapted.
Best regards
